examples/read_write_mutex/source/app.d

```diff
@@ -0,0 +1,103 @@
+
+import vibe.d;
+
+interface MyBlockingRestInterface {
+	@path("/")
+	int getIndex();
+}
+
+class RestInterfaceImplementation : MyBlockingRestInterface {
+	private {
+		TaskReadWriteMutex m_mutex;
+	}
+
+	this()
+	{
+		m_mutex = new TaskReadWriteMutex(TaskReadWriteMutex.Policy.PREFER_WRITERS);
+	}
+
+	//This method naively simulates access to a shared resource protected
+	//by a TaskReadWriteMutex. Read operations are performed frequently (95%) 
+	//and finish quickly (1ms wait time), while write operations occur rarely
+	//(5%) and take much longer to process (50ms wait time).
+	int getIndex()
+	{
+	    import std.random;
+	    import std.stdio;
+	    auto i = uniform(0,20);
+
+	    if (i == 0) //In a rare case, lock for writing
+	    {
+	        synchronized(m_mutex.writer)
+	        {
+	        	version(PrintRequests)
+	        		writeln("Blocked for writing!");
+	        	//Simulate a slow operation
+	            vibe.core.core.sleep(100.msecs);
+	        }
+	    }
+	    else //More regularly, lock for reading
+	    {
+	        synchronized(m_mutex.reader)
+	        {
+	        	version(PrintRequests)
+	        		writeln("Blocked for reading!");
+	        	//Simulate a faster operation
+	            vibe.core.core.sleep(1.msecs);
+	        }
+	    }
+	    return 0;
+	}
+}
+
+
+__gshared {
+    TaskMutex s_taskMutex;
+    TaskCondition s_taskCondition;
+    ulong s_runningTasks;
+}
+
+shared static this()
+{
+    s_taskMutex = new TaskMutex();
+    s_taskCondition = new TaskCondition(s_taskMutex);
+
+	auto routes = new URLRouter;
+	registerRestInterface(routes, new RestInterfaceImplementation());
+
+	auto settings = new HTTPServerSettings();
+	settings.port = 8080;
+	settings.options = HTTPServerOption.parseURL | HTTPServerOption.distribute;
+	settings.bindAddresses = ["::1", "127.0.0.1"];
+
+	listenHTTP(settings, routes);
+
+	//Wait for a couple of seconds for the server to be initialized properly and then start
+	//multiple concurrent threads that simultaneously start queries on the rest interface defined above.
+	setTimer(5.seconds, {
+
+		scope(exit)
+			exitEventLoop(true);
+
+        //Start multiple tasks performing requests on "http://localhost:8080/" concurrently
+        synchronized(s_taskMutex) s_runningTasks = workerThreadCount();
```

Comment:
I added the counter initialization with `workerThreadCount()` and did a rebase to the upstream master branch. Interestingly, `workerThreadCount()` even worked before the rebase (at least on my system).
I'm quite sure the `synchronized(s_taskMutex)` is not technically necessary, but since this line sets up the condition variable, I figured it doesn't hurt to have it there.